modules/calcite/src/main/java/org/apache/ignite/internal/processors/query/calcite/exec/ExecutionServiceImpl.java

```diff
@@ -1149,4 +1165,80 @@ private void onError(Throwable error) {
             tryClose();
         }
     }
+
+    /** */
+    private static class FixDependentInsertNodeShuttle extends IgniteRelShuttle {
```

Comment:
Let's rename the class (and update javadocs), since it's already not only for insert node

Reply:
fixed. I've also added a javadoc describing problem this shuttle is supposed to fix.

Reply:
There still some "insert node" phrases in javadocs.
About "update" statement: "insert" statement as select from the same table it's a rare case, and memory overconsumption caused by table spool not so important here, but "update" statement will always be depended on the scan of the updated table and table spool always will be used. But for updates, it's not always needed. It's needed only if we modify columns used in the index we are iterating. Perhaps it's worth to include such optimization. WDYT?

Reply:
> it's worth to include such optimization

Sounds reasonable. I'll think how it would look like

Reply:
Fixed